htdocs/product/class/html.formproduct.class.php

```diff
@@ -655,13 +655,11 @@ public function selectLotDataList($htmlname = 'batch_id', $empty = 0, $fk_produc
 		}
 
 		foreach ($productIdArray as $productId) {
-			if (array_key_exists($productId, $this->cache_lot)) {
-				foreach ($this->cache_lot[$productId] as $id => $arraytypes) {
-					if (empty($fk_entrepot) || $fk_entrepot == $arraytypes['entrepot_id']) {
-						$label = $arraytypes['entrepot_label'] . ' - ';
-						$label .= $arraytypes['batch'];
-						$out .= '<option>' . $arraytypes['batch'] . '</option>';
-					}
+			foreach ($this->cache_lot[$productId] as $id => $arraytypes) {
```

Comment:
The test if (array_key_exists) required for php8 compatibility seems missing. Can you restore it ?

Reply:
@eldy Done